indexing-service/src/main/java/io/druid/indexing/common/task/HadoopTask.java

```diff
@@ -102,23 +101,16 @@ protected ClassLoader buildClassLoader(final TaskToolbox toolbox) throws Excepti
                                                           ? hadoopDependencyCoordinates
                                                           : toolbox.getConfig().getDefaultHadoopCoordinates();
 
-    final List<URL> extensionURLs = Lists.newArrayList();
+    final List<URL> jobURLs = Lists.newArrayList(
+        Arrays.asList(((URLClassLoader) HadoopIndexTask.class.getClassLoader()).getURLs())
+    );
+
     for (final File extension : Initialization.getExtensionFilesToLoad(extensionsConfig)) {
       final ClassLoader extensionLoader = Initialization.getClassLoaderForExtension(extension);
-      extensionURLs.addAll(Arrays.asList(((URLClassLoader) extensionLoader).getURLs()));
+      jobURLs.addAll(Arrays.asList(((URLClassLoader) extensionLoader).getURLs()));
     }
 
-    final List<URL> nonHadoopURLs = Lists.newArrayList();
-    nonHadoopURLs.addAll(Arrays.asList(((URLClassLoader) HadoopIndexTask.class.getClassLoader()).getURLs()));
-
-    final List<URL> druidURLs = ImmutableList.copyOf(
-        Collections2.filter(nonHadoopURLs, IS_DRUID_URL)
```

Comment:
This moves the hadoop coordinates last. The IS_DRUID_URL thing was used in trying to make the ordering on the classpath very explicit, in which case druid classes would come "after" hadoop classes. But this is not absolutely required since they are all in the same classloader.

Jars with a lower index in the resulting list will get preference on loading classes, and one of the changes here is to make the hadoop coordinate LAST. So that any non-hadoop classes take precedence *LOCALLY*. It does not change the jar handling in the hadoop job itself.